### ISSUE TITLE
Analyzer for 'Reported diagnostic with ID 'XXX' is not supported by the analyzer.' does not work with property initializers

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn-analyzers#1689]
#### Analyzer package
Microsoft.CodeAnalysis.Analyzers

#### Repro steps

Warn if the analyzer reports diagnostic for a descriptor that is not in `SupportedDiagnostics`
```cs
namespace RoslynSandbox
{
    using System.Collections.Immutable;
    using Microsoft.CodeAnalysis;
    using Microsoft.CodeAnalysis.CSharp;
    using Microsoft.CodeAnalysis.Diagnostics;

    [DiagnosticAnalyzer(LanguageNames.CSharp)]
    internal class FooAnalyzer : DiagnosticAnalyzer
    {
        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(
            FooInfo.Descriptor);

        public override void Initialize(AnalysisContext context)
        {
            context.RegisterSyntaxNodeAction(
                c => c.ReportDiagnostic(
                    Diagnostic.Create(
                        BarInfo.Descriptor,
                        c.Node.GetLocation())),
                SyntaxKind.AbstractKeyword);
        }
    }

    internal static class FooInfo
    {
        public const string DiagnosticId = "FOO";

        internal static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
            id: DiagnosticId,
            title: "",
            messageFormat: "",
            category: "",
            defaultSeverity: DiagnosticSeverity.Warning,
            isEnabledByDefault: true,
            description: "");
    }

    internal static class BarInfo
    {
        public const string DiagnosticId = "BAR";

        internal static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
            id: DiagnosticId,
            title: "",
            messageFormat: "",
            category: "",
            defaultSeverity: DiagnosticSeverity.Warning,
            isEnabledByDefault: true,
            description: "");
    }
}
```

**Comment by mavasani** [comment]
@JohanLarsson We already have [RS1005](https://github.com/dotnet/roslyn-analyzers/blob/26fe065161a871718c717da1821d82c4decbc8e9/src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/ReportDiagnosticAnalyzer.cs#L25) in Microsoft.CodeAnalysis.Analyzers but does not seem to work for property initializers. If you replace the following:

```
public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(FooInfo.Descriptor);
```

with 

```
public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(FooInfo.Descriptor);
```

RS1005 is indeed reported. I will send a fix to handle this case in the analyzer.
